SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/DynamicActorDormantAllChangePropertyTest.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "DynamicActorDormantAllChangePropertyTest.h"
+
+#include "DormancyTestActor.h"
+#include "EngineClasses/SpatialNetDriver.h"
+
+// The test checks that modifying a replicated property on an actor that has NetDormancy DORM_DormantAll will result in the property not
+// being replicated.
+
+ADynamicActorDormantAllChangePropertyTest::ADynamicActorDormantAllChangePropertyTest()
+{
+	Author = "Matthew Sandford";
+}
+
+void ADynamicActorDormantAllChangePropertyTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Step 1 - Spawn dormancy actor and check NetDormancy is DORM_DormantAll
+	AddStep(TEXT("ServerSpawnDormancyActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AActor* Actor = CreateDormancyTestActor();
+		Actor->SetNetDormancy(DORM_DormantAll);
+		FinishStep();
+	});
+
+	// Step 2 - Wait for actor channel to be ReadyForDormancy
+	AddStep(
+		TEXT("ServerWaitForActorChannelReadyForDormancy"), FWorkerDefinition::Server(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			bool bIsReadyForDormancy = true;
+
+			UWorld* World = GetWorld();
+			USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(World->GetNetDriver());
+			if (SpatialNetDriver != nullptr)
+			{
+				for (TActorIterator<ADormancyTestActor> Iter(GetWorld()); Iter; ++Iter)
+				{
+					const ADormancyTestActor* DormancyTestActor = *Iter;
+					if (DormancyTestActor != nullptr)
```

Comment:
maybe a little bit more strict here? Like an `AssertTrue ( != nullptr)`?

Reply:
and making sure the loop executes?

Reply:
Also is the wait for `bIsReadyForDormancy` important for further functionality in the test or is this a test of the SpatialNetDriver's internals?

Reply:
This is important for functionality of the test rather than testing the netdriver.

Reply:
Have added a count requirement.